src/screens/OnboardingFlow.tsx

```
@@ -44,27 +47,61 @@ export function OnboardingFlow() {
 
   const handleSkip = () => {
     setCurrentStep(INTRO_STEPS.length) // Jump to setup
   }
 
+  const onViewableItemsChanged = useRef(
+    ({ viewableItems }: { viewableItems: ViewToken<(typeof INTRO_STEPS)[number]>[] }) => {
+      if (viewableItems.length > 0 && viewableItems[0].index !== null) {
+        setCurrentStep(viewableItems[0].index)
+      }
+    },
+  ).current
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/screens/OnboardingFlow.tsx:67:34
  65 |   }
  66 |
> 67 |   const onViewableItemsChanged = useRef(
     |                                  ^^^^^^^
> 68 |     ({ viewableItems }: { viewableItems: ViewToken<(typeof INTRO_STEPS)[number]>[] }) => {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 69 |       if (viewableItems.length > 0 && viewableItems[0].index !== null) {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 70 |         setCurrentStep(viewableItems[0].index)
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 71 |       }
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 72 |     },
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 73 |   ).current
     | ^^^^^^^^^^^^ Cannot access ref value during render
  74 |
  75 |   const viewabilityConfig = useRef({
  76 |     itemVisiblePercentThreshold: 50,

(react-hooks/refs)